File: src/Aardvark.OpenCV.Tests/FittingTests.cs
using Aardvark.Base;
using NUnit.Framework;

namespace Aardvark.OpenCV.Tests
{
    [TestFixture]
    internal class FittingTests
    {
        private static V3d Vec(float x, float y, float z) => new(x, y, z);

        [OneTimeSetUp]
        public void Init()
        {
            Aardvark.Base.Aardvark.Init();
        }

        [Test]
        public void FitPlane3dLeastSquares()
        {
            V3d[] data = { Vec(1, 2, -1), Vec(5, 4, -1), Vec(23, 100, -1) };
            var plane = data.FitPlane3dLeastSquares();
            Assert.AreEqual(new Plane3d(V3d.ZAxis, -1), plane);
        }
    }
}

File: src/Aardvark.OpenCV.Tests/Tests/FittingTests.fs
namespace Aardvark.OpenCV.Tests

open Aardvark.Base
open Aardvark.OpenCV
open NUnit.Framework
open FsUnit

module ``Fitting Tests`` =

    [<SetUp>]
    let init() =
        Aardvark.Init()

    [<Test>]
    let ``Plane3d least squares``() =
        let data = [| V3d(1.0, 2.0, -1.0); V3d(5.0, 4.0, -1.0); V3d(23.0, 100.0, -1.0) |]
        let plane = data.FitPlane3dLeastSquares();
        plane |> should equal (Plane3d(V3d.ZAxis, -1))